force-app/nksHomePage/lwc/nksHomePageList/nksHomePageList.js

```diff
@@ -285,63 +289,43 @@ export default class nksHomePageList extends NavigationMixin(LightningElement) {
         return Object.keys(this.subscription).length !== 0 && this.subscription.constructor === Object;
     }
 
-    get newsRecords() {
-        let sortedList = [];
-        let recordsToSort = [];
 
+    get newsRecords() {
         if (this.isNews && this.records && Array.isArray(this.records)) {
-            recordsToSort = JSON.parse(JSON.stringify(this.records));
-            sortedList = recordsToSort.sort(function (x, y) {
-                let index = 0;
-
-                // pinned items first
-                if (x.pin === y.pin) {
-                    index = 0;
-                } else {
-                    if (x.pin === true) {
-                        index = -1;
-                    } else {
-                        index = 1;
-                    }
-                }
-                return index;
-            });
+            let sortedList = [...this.records];
+    
+            // Sort by 'pin' property in descending order (pinned items first)
+            sortedList.sort((a, b) => (b.pin === a.pin ? 0 : b.pin ? 1 : -1));
+            return sortedList;
         }
-        return sortedList;
+        return [];
     }
 
     get isNews() {
-        if (this.objectName === 'NKS_Announcement__c' && this.filter && this.filter.includes('News')) {
-            return true;
-        }
-        return false;
+        return this.objectName === 'NKS_Announcement__c' && this.filter && this.filter.includes('News');
```

Comment:
Her this.filter kan gjerne fjernes (man får varsel fra sonarLint): 
`        return this.objectName === 'NKS_Announcement__c' && this.filter?.includes('News');
`